perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java

```diff
@@ -5902,6 +5901,108 @@ public void getGroupsPage_ParentGroup_ReturnsAttributes () throws Exception {
+
+		List<String> attributeNames = new ArrayList<>();
+		Map<UUID, List<Attribute>> memberGroupAttributes = new HashMap<>();
+
+		AttributeDefinition attr = new AttributeDefinition();
+		attr.setNamespace(AttributesManager.NS_MEMBER_GROUP_ATTR_VIRT);
+		attr.setFriendlyName("groupStatus");
+		attr.setDisplayName("Group membership status");
+		attr.setType(String.class.getName());
+		attr.setDescription("Whether member is VALID or EXPIRED in a group.");
+
+		attributesManager.createAttribute(sess, attr);
+
+		attributeNames.add(new Attribute(attr).getName());
+		attributeNames.add(AttributesManager.NS_MEMBER_GROUP_ATTR_DEF +":groupMembershipExpiration");
+
+		memberGroupAttributes.put(group.getUuid(), new ArrayList<>());
+		memberGroupAttributes.get(group.getUuid()).add(attributesManager.getAttribute(sess, member, group, attr.getName()));
+		memberGroupAttributes.get(group.getUuid()).add(attributesManager.getAttribute(sess, member, group, AttributesManager.NS_MEMBER_GROUP_ATTR_DEF +":groupMembershipExpiration"));
+
+		memberGroupAttributes.put(group2.getUuid(), new ArrayList<>());
+		memberGroupAttributes.get(group2.getUuid()).add(attributesManager.getAttribute(sess, member, group2, attr.getName()));
+		memberGroupAttributes.get(group2.getUuid()).add(attributesManager.getAttribute(sess, member, group2, AttributesManager.NS_MEMBER_GROUP_ATTR_DEF +":groupMembershipExpiration"));
+
+		memberGroupAttributes.put(groupsManagerBl.getGroupByName(sess, vo, "members").getUuid(), new ArrayList<>());
+		memberGroupAttributes.get(groupsManagerBl.getGroupByName(sess, vo, "members").getUuid()).add(attributesManager.getAttribute(sess, member, groupsManagerBl.getGroupByName(sess, vo, "members"), attr.getName()));
+		memberGroupAttributes.get(groupsManagerBl.getGroupByName(sess, vo, "members").getUuid()).add(attributesManager.getAttribute(sess, member, groupsManagerBl.getGroupByName(sess, vo, "members"), AttributesManager.NS_MEMBER_GROUP_ATTR_DEF +":groupMembershipExpiration"));
+
+
+		GroupsPageQuery query = new GroupsPageQuery(10, 0, SortingOrder.ASCENDING, GroupsOrderColumn.ID, "", member.getId());
+		Paginated<RichGroup> groups = groupsManager.getGroupsPage(sess, vo, query, attributeNames);
+
+		assertNotNull(groups);
+		assertEquals(groups.getData().size(), 3);
+
+		for (RichGroup g : groups.getData()) {
+			assertTrue(g.getAttributes().containsAll(memberGroupAttributes.get(g.getUuid())));
+		}
+	}
+
+	@Test
+	public void getGroupsPageForMemberWithAllAttributes() throws Exception {
+
+		vo = setUpVo();
+
+		perun.getGroupsManager().createGroup(sess, vo, group);
+		perun.getGroupsManager().createGroup(sess, vo, group2);
+		perun.getGroupsManager().createGroup(sess, vo, group3);
+
+		Member member = setUpMember(vo);
+
+		groupsManagerBl.addMember(sess, group, member);
+		groupsManagerBl.addMember(sess, group2, member);
+
+		Map<UUID, List<Attribute>> attributes = new HashMap<>();
+
+		AttributeDefinition attr = new AttributeDefinition();
```

Comment:
It would be nice to have the repetitive code extracted to private method. There's also `setUpMemberGroupAttributes` you can use.